### PR TITLE
[Runtimes] Add validation on default handler type

### PR DESCRIPTION
**Pull request by Hedingber**
Fixes https://jira.iguazeng.com/browse/ML-876

### CODE REVIEW

File: mlrun/model.py [hunk 1]
```diff
@@ -48,6 +49,14 @@ def _verify_dict(param, name, new_type=None):
             return new_type.from_dict(param)
         return param
 
+    @staticmethod
+    def _verify_str(param, name):
+        if param is not None and not isinstance(param, str):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Parameter {name} must be a string. got {type(param)}"
```
Comment (eran-nussbaum) [inline mlrun/model.py:56]: Should capitalize the first word in a sentence.
```suggestion
                f"Parameter {name} must be a string. Got {type(param)}"
```

File: tests/api/runtimes/test_pod.py [hunk 2]
```diff
@@ -50,3 +50,24 @@ def test_with_limits_regex_validation(self):
...
+        itself to the handler kwarg, instead of giving the function name (string)
+        This test is here to verify that the failure is fast and clear
+        """
+
+        def some_func():
+            pass
+
+        with pytest.raises(
+            mlrun.errors.MLRunInvalidArgumentError,
+            match=r"Parameter default_handler must be a string. got <class 'function'>",
```
Comment (eran-nussbaum) [inline tests/api/runtimes/test_pod.py:66]: Same as last comment.
```suggestion
            match=r"Parameter default_handler must be a string. Got <class 'function'>",
```

**Comment by liranbg** [comment]
@Tankilevitch  - this is stale. reopen with up to date code

**Comment by eran-nussbaum** [comment]
You _better_ merge it!

**Pull request closed without merging**